deepspeed/runtime/engine.py

```diff
@@ -2868,11 +2868,17 @@ def _load_zero_checkpoint(self, load_dir, tag, load_optimizer_states=True):
             if zero_sd_list is None:
                 return False
 
-        self.optimizer.load_state_dict(state_dict_list=zero_sd_list,
-                                       load_optimizer_states=load_optimizer_states,
-                                       load_from_fp32_weights=self.zero_load_from_fp32_weights(),
-                                       checkpoint_folder=checkpoint_folder,
-                                       load_serial=load_serial)
+        if self.zero_optimization_stage() == 3:
+            self.optimizer.load_state_dict(state_dict_list=zero_sd_list,
```

Comment:
The preferred solution here is to add `load_serial` to bf16_optimizer's `load_state_dict()`. Our intention is to propagate that feature to all the optimizer implementations.

Reply:
@ShijieZZZZ, FYI